### PR TITLE
adding notification contextualisation support

### PR DESCRIPTION
**Pull request by neil-stephen-mcgonigle**
contextualised template ids used in place of body content.

proposed version bump 6.0.0? breaking changes due to task change from array to map?

### CODE REVIEW

**Comment by neil-stephen-mcgonigle** [comment]
Having a think about version bump

**Review by armoj — CHANGES_REQUESTED** [review]

File: src/template.js [hunk 1]
```diff
@@ -543,26 +543,80 @@ module.exports = {
     meta: {
         questionnaireDocumentVersion: '3.0.0',
         onComplete: {
-            tasks: [
-                {
+            tasks: {
+                sendEmail: {
```
[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply src/template.js:547]: https://github.com/CriminalInjuriesCompensationAuthority/q-templates-application/blob/72ec67042ab22ff3592e06c9ccfdd7bb33d3170c/index.js#L13339

[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply src/template.js:547]: My branch is built upon your original notifications branch, for reference.

File: src/lib/deleteKeys.js [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 function deleteKeys(obj, unwantedKeyArray) {
     const result = obj;
+    // meta tasks do not have unwantedKeys at the moment
```
[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply src/lib/deleteKeys.js:5]: neither do I, but I didn't write it :)

File: src/lib/deleteKeys.test.js [hunk 3]
```diff
@@ -77,4 +78,16 @@ describe('delete keys', () => {
 
         expect(actual).toEqual(expected);
     });
+
+    it('should strip the current templaate of the provided nested keys and return expected properties', () => {
```
[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply src/lib/deleteKeys.test.js:82]: fixed

File: src/template.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply src/template.js:547]: leaving this as is for now.

File: src/lib/deleteKeys.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply src/lib/deleteKeys.js]: added explicit null and undefined checks

**Review by BarryPiccinni — APPROVED** [review]
Approved, but it's a bit of a concern that you were able to make this change and not have to update the q-schema module. The validation of the `meta` section is probably too loose.

**Comment by armoj** [comment]
is [this commit](e5d6b4e4f43a804090ba3993e8d24d00e1e381c6) now redundant due to Barry's updates?

**Pull request merged**